hopper/flash_api.cpp

```diff
@@ -1646,23 +1719,164 @@ mha_combine(at::Tensor out_partial,         // num_splits x batch_size x seqlen
     params.o_row_stride = out.stride(1);
     params.o_head_stride = out.stride(2);
     params.o_batch_stride = out.stride(0);
-    params.arch = at::cuda::getCurrentDeviceProperties()->major * 10 + at::cuda::getCurrentDeviceProperties()->minor;
+    params.arch = dprops->major * 10 + dprops->minor;
 
     if (seqlen > 0 && batch_size > 0) {
-        auto stream = at::cuda::getCurrentCUDAStream().stream();
+        auto device_idx = torch::stable::accelerator::getCurrentDeviceIndex();
+        auto stream = (cudaStream_t)torch::stable::accelerator::getCurrentStream(device_idx).id();
         run_mha_fwd_combine(params, stream, false /*enable_pdl*/);
     }
 
-    at::Tensor out_padded = out;
+    Tensor out_padded = out;
     if (head_size_og % alignment != 0) {
-        out = out.index({"...", torch::indexing::Slice(torch::indexing::None, head_size_og)});
+        out = torch::stable::narrow(out, -1, 0, head_size_og);
         // if (out_.has_value()) { out_.value().copy_(out); }
     }
 
     return {out, softmax_lse};
 }
 
-TORCH_LIBRARY(flash_attn_3, m) {
+void boxed_mha_fwd(
```

Comment:
I don't quite understand why we need the "boxed" version

Reply:
This is a limitation of `STABLE_TORCH_LIBRARY` for now where it only accepts the boxed kernel. But echoing [Jane's comment above](https://github.com/Dao-AILab/flash-attention/pull/1791#discussion_r2301938634) above 

> We do want to eventually automate this and hide it from the user completely! We ultimately want to support this in the dispatcher which requires a bit of a lift as we support more IValues.